halfshell/source_s3.go

```diff
@@ -69,7 +69,9 @@ func (s *S3ImageSource) GetImage(request *ImageSourceOptions) (*Image, error) {
 }
 
 func (s *S3ImageSource) signedHTTPRequestForRequest(request *ImageSourceOptions) *http.Request {
-	imageURLPathComponents := strings.Split(request.Path, "/")
+	path := s.Config.PathPrefix + request.Path
```

Comment:
Want to update the PR and I'll merge? Also, can you please make the relevant updates to the README?